### PR TITLE
Add RTL support to progress-bar

### PR DESCRIPTION
**Pull request by negreirosleo**
#### What does this PR do?

Adds RTL support to the progress meter,

it works by passing the `rtl` prop to meter:
```js
  <Meter type="bar" rtl value={value} />
```

Or if the `dir` prop on `theme` has the 'rtl' value:

```js
    <Grommet dir="rtl">
      <Box align="center" pad="large">
        <Meter type="bar" value={value} />
      </Box>
    </Grommet>
```

#### What testing has been done on this PR?

A story was written for both cases

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/6358

#### Screenshots (if appropriate)

image

#### Do the grommet docs need to be updated?

Yes, the `rtl` prop should be added to the docs

#### Should this PR be mentioned in the release notes?

Yes

#### Is this change backwards compatible or is it a breaking change?

It is backwards compatible.

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
Nice start on this! A few thoughts...
I am thinking we should allow the direction of meter to be reversed when it is a circle or semicircle not just a bar.
Also I am wondering if we should call the prop `reverse` instead of `rtl`. There could be cases where people want to reverse the direction even with using left to right orientated languages. `reverse` makes the meaning of prop a bit more clear. Anyone using it for right to left purposes will likely set that at the Grommet component level

**Comment by negreirosleo** [comment]
> Nice start on this! A few thoughts... I am thinking we should allow the direction of meter to be reversed when it is a circle or semicircle not just a bar. Also I am wondering if we should call the prop `reverse` instead of `rtl`. There could be cases where people want to reverse the direction even with using left to right orientated languages. `reverse` makes the meaning of prop a bit more clear. Anyone using it for right to left purposes will likely set that at the Grommet component level

In the case of the semicircle, it should rotate counter clockwise right?
also, should we also add it for the vertical direction and remove the dir="rtl" check from the theme?

**Comment by jcfilben** [comment]
> In the case of the semicircle, it should rotate counter clockwise right?

Yes, that's what I'm thinking

> also, should we also add it for the vertical direction and remove the dir="rtl" check from the theme?

I'm a little torn on if we should add it for the vertical direction. I can't think of a use case for someone needing this so maybe we should hold off on adding it for the vertical direction. And setting `dir="rtl"` shouldn't affect the vertical direction meter.

Here is a little more detail of what I'm thinking:
Add a `reverse` prop on the `<Meter>` component. The we leave the `dir="rtl"` as is on the `<Grommet>` component.
When `reverse=true` or `dir="rtl"` the meter should start on the right and move to the left or counterclockwise for circle/semicircle meters.
If `reverse=true` and `dir="rtl"` the meter should start on the left and move to the right or clockwise for circle/semicircle meters.

**Comment by jcfilben** [comment]
Hey @negreirosleo just checking in on this PR to see if you had any questions on this or if you are still interested in working on this

**Comment by negreirosleo** [comment]
@jcfilben sorry taking so long! but I applied the suggestions!

**Comment by jcfilben** [comment]
> @jcfilben sorry taking so long! but I applied the suggestions!

Thanks! I'll take a closer look at the changes tomorrow, but I took a quick scan through the stories and things are looking good there!

**Review by jcfilben — COMMENTED** [review]
Logic all looks good! I think we just need one storybook story to show the reverse prop. Can the ReverseBarTheme and ReverseCircle stories be removed and unit tests be added for these cases?

**Comment by negreirosleo** [comment]
@jcfilben I'm having a issue where most of the snapshots are failling locally but the one that is failling on the CI is passing here, do you know what this can be?

**Comment by jcfilben** [comment]
> @jcfilben I'm having a issue where most of the snapshots are failing locally but the one that is failling on the CI is passing here, do you know what this can be?

Hey @negreirosleo can you try updating your branch with master locally and then running `rm -rf yarn.lock && rm -rf node_modules && yarn cache clean && yarn install` and then see if the snapshots are still failing? If they are still failing after this and you are using node v18 you can try using node v16

**Comment by negreirosleo** [comment]
that was it, thanks @jcfilben !!

**Review by jcfilben — APPROVED** [review]
Thanks for making the changes, this looks good to me!

File: src/js/components/Meter/Meter.js [hunk 1]
```diff
@@ -25,18 +27,26 @@ const Meter = forwardRef(
       thickness = 'medium',
       type = 'bar',
       value,
+      reverse,
```
Comment (taysea) [inline src/js/components/Meter/Meter.js:30]:
```suggestion
      reverse: reverseProp,
```

File: src/js/components/Meter/Meter.js [hunk 2]
```diff
@@ -25,18 +27,26 @@ const Meter = forwardRef(
...
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+
     // normalize values to an array of objects
     const values = useMemo(() => {
       if (valuesProp) return valuesProp;
       if (value) return [{ color, value }];
       return [];
     }, [color, value, valuesProp]);
 
+    const reverseDirection =
```
Comment (taysea) [inline src/js/components/Meter/Meter.js:45]:
```suggestion
    const reverse =
```

File: src/js/components/Meter/Meter.js [hunk 3]
```diff
@@ -25,18 +27,26 @@ const Meter = forwardRef(
...
+      direction === 'horizontal' &&
+      (theme.dir === 'rtl' || reverse) &&
```
Comment (taysea) [inline src/js/components/Meter/Meter.js:47]:
```suggestion
      (theme.dir === 'rtl' || reverseProp) &&
```

File: src/js/components/Meter/Meter.js [hunk 4]
```diff
@@ -25,18 +27,26 @@ const Meter = forwardRef(
...
+      !(theme.dir === 'rtl' && reverse);
```
Comment (taysea) [inline src/js/components/Meter/Meter.js:48]:
```suggestion
      !(theme.dir === 'rtl' && reverseProp);
```

File: src/js/components/Meter/Meter.js [hunk 5]
```diff
@@ -49,6 +59,7 @@ const Meter = forwardRef(
           thickness={thickness}
           background={background}
           direction={direction}
+          reverse={reverseDirection}
```
Comment (taysea) [inline src/js/components/Meter/Meter.js:62]:
```suggestion
          reverse={reverse}
```

File: src/js/components/Meter/Meter.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Meter/Meter.js:30]: Some small naming/code convention suggestions to align with how we handle in other places.

Comment (taysea) [reply src/js/components/Meter/Meter.js:30]: Can we also move the placement of where this is destructured to be in alphabetical order?

File: src/js/components/Meter/index.d.ts [hunk 6]
```diff
@@ -16,6 +16,7 @@ export interface MeterProps {
   gridArea?: GridAreaType;
   margin?: MarginType;
   max?: number;
+  reverse?: boolean;
```
Comment (taysea) [inline src/js/components/Meter/index.d.ts:19]: Do we also need to add this to the `propTypes.js` file for Meter?

Comment (negreirosleo) [reply src/js/components/Meter/index.d.ts:19]: I makes sense adding it, so I added!

**Comment by negreirosleo** [comment]
@taysea I applied your suggestions :+1:

**Review by taysea — APPROVED** [review]
Nice work!

**Pull request merged**